emulators/mame/Portfile

```diff
@@ -41,20 +126,45 @@ depends_lib-append  \
                     port:sqlite3 \
                     port:zlib
 
+# Note: Port 'coreutils' needed for 'grealpath', used in launcher script.
+depends_run-append  \
+                    port:coreutils
+
+#------------------------------------------------------------------------------
+# Build Options
+#------------------------------------------------------------------------------
+
+use_parallel_build  yes
+
 use_configure       no
 
+# https://github.com/mamedev/mame/issues/6004
+compiler.blacklist  {clang < 1000}
+
 # https://github.com/mamedev/mame/issues/3788
-if {${os.platform} eq "darwin" && ([vercmp $xcodeversion 9.0] < 0)} {
-    configure.cxxflags-append -DMAME_DEVCB_GNUC_BROKEN_FRIEND
-}
+# As long as we're blacklisting clangs from the affected Xcode versions,
+# we don't need to add this flag.
+#if {${os.platform} eq "darwin" && ([vercmp $xcodeversion 9.0] < 0)} {
+#    configure.cxxflags-append -DMAME_DEVCB_GNUC_BROKEN_FRIEND
+#}
 
 compiler.cxx_standard   2014
 
+# Note: Compiler optimization level is controlled by makefile variable 'OPTIMIZE', below.
+configure.cxxflags  -stdlib=libc++
```

Comment:
Use the `configure.cxx_stdlib` variable, and only append this flag for compilers that understand it (clang). Append to the existing flags so as not to wipe out the flags MacPorts already sets. In fact, MacPorts already puts this flag in `configure.cxxflags` for you, so just remove this entirely?
```suggestion
```

Reply:
Great feedback!

Currently verifying this locally, to be sure the resultant build flags are correct.

Reply:
Okay, now I remember why I replaced the standard flags: The compiler optimization level is included, which we don't want. That's because the Mame makefile controls that, via make variable 'OPTIMIZE'.

A few ports use the following to deal with that, which covers the four most common cases:
`configure.cxxflags-delete  -O1 -O2 -O3 -Os
`

Given that it's reasonably non-fragile, I'll combine the latter, with your earlier suggested change.

Reply:
Changes committed.